cpp/fetcher/remote_peer.cc

```diff
@@ -67,35 +68,37 @@ void RemotePeer::Impl::DoneGetSTH(
     return;
   }
 
-  // TODO(pphaneuf): At least some of the CHECK-failing here might
-  // have to be changed to... something else.
-  CHECK_EQ(verifier_->VerifySignedTreeHead(
-               *new_sth, 0, (time(NULL) + 10) * kNumMillisPerSecond),
-           LogVerifier::VERIFY_OK)
-      << "could not validate STH:\n" << new_sth->DebugString();
-
-  // TODO(alcutter): Need to check STH consistency here, including for older
-  // STHs which might come in.
-
-  lock_guard<mutex> lock(lock_);
-
-  if (new_sth->tree_size() < 0) {
-    LOG(WARNING) << "Unexpected tree size in new_sth:\n"
-                 << new_sth->DebugString();
-  } else if (sth_ && new_sth->tree_size() < sth_->tree_size()) {
-    LOG(WARNING) << "Received old STH:\n" << new_sth->DebugString();
-  } else if (!sth_ || new_sth->timestamp() > sth_->timestamp()) {
-    // This STH is good, we'll take it.
-    sth_ = new_sth;
-    if (on_new_sth_) {
-      on_new_sth_(*sth_);
+  if (status == AsyncLogClient::OK) {
```

Comment:
What's the warning that this fixed? It makes sense, somewhat, but I'm really curious of what turned this up!